### ISSUE TITLE
[Form] Add support for Bootstrap 4 forms

### ISSUE THREAD

**Issue by javiereguiluz** [symfony/symfony#16289]
Bootstrap 4 is still in alpha but we should prepare for when it's released (no specific date yet). Regarding the form component, changes from Bootstrap 3 are small, but nonetheless important.

![bootstrap_4_forms](https://cloud.githubusercontent.com/assets/73419/10580501/0f1097d6-767e-11e5-8a02-464e72c493ad.png)

Source: http://v4-alpha.getbootstrap.com/migration/#forms

**Comment by FractalizeR** [comment]
I think this is incomplete. I tried this, but it looks like form errors are no longer printed in red for example. It can be achieved using `text-danger` style for example.

**Comment by decebaldecebal** [comment]
Just noticed this and in my opinion for this to be complete it should also include a layout where the custom Bootstrap 4 elements are used.

**Comment by szymach** [comment]
But Bootstrap4 is still in alpha stage, is there a point introducing a template that may simply stop working after another alpha release?

**Comment by hiddewie** [comment]
There will be a beta release very soon, see https://github.com/twbs/bootstrap/projects/3.

**Comment by szymach** [comment]
Well that still may change to various reasons, it would not be uncommon for release date to be postponed. Anyhow, I think it would be better to wait until it starts approaching a stable release, since we have no idea how much it could change.

**Comment by PhilETaylor** [comment]
Bump - cough - Currently v4.0.0-beta http://getbootstrap.com/

**Comment by ReSpawN** [comment]
We're currently developing a narrowcasting solution with Symfony 3.3.6 and Boostrap 4 beta, and yes, the form styles are currently completely messed up due to all the changes. I'm eagerly looking forward to stable support. Supporting a beta however normally isn't something I would do, but Bootstrap has proven time and time again to be one of the most stable of frontend framework solutions, mainly taking CSS.
How far along is your PR @hiddewie ? I might implement yours before week end.

**Comment by hiddewie** [comment]
@ReSpawN Thank you for the interest. 

The PR should be very functional for the BS 4 beta. I am using it in small personal projects myself, but I have not seen it implemented in large scale projects. The only thing that I was able to determine which does not work is a form widget with invalid state and invalid text for (sets of) radio buttons and checkboxes. I've raised this issuse with the Bootstrap team (twbs/bootstrap#23426).
